pkg/operator/ceph/cluster/mon/mon.go

```diff
@@ -413,6 +413,7 @@ func realScheduleMonitor(c *Cluster, mon *monConfig) (SchedulingResult, error) {
 	d := c.makeDeployment(mon)
 	d.Name += "-canary"
 	d.Spec.Template.ObjectMeta.Name += "-canary"
+	d.Labels["mon_type"] = "canary"
```

Comment:
I think this label should be applied to both the deployment and the deployment's pod template.

Reply:
I do not understands well @BlaineEXE. Do you mean, to include this labels directly in the cluster "makeDeployment" method?

Reply:
This is only putting labels on the deployment. The deployment's pod spec template should (I think) also get the label.

Reply:
I think that is better to leave the code in the spec.go file only for management of "normal" monitors. 
In the realScheduleMonitor function we are labeling the canary monitors only. Normal monitors do not need the "mon_type" label or the "canary_pvc_name". 
What you suggest could be useful if in the case that we have more that "normal" and "canary" mons.

Reply:
I think there is a misunderstanding somewhere. Kubernetes deployments have labels, and the pods they create have a separate spec for labels. If we are labeling the canary deployments, we should also label the deployment's pods via the pod spec.

Reply:
How about naming the label like this? It seems more natural.
```
d.Labels["mon_canary"] = "true"
```